modules/fedimint-gw-client/src/receive.rs

```diff
@@ -0,0 +1,15 @@
+/// State machine that handles an intercepted HTLC from the Lightning node.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    HTLCIntercepted -- await preimage offer accepted --> BuyPreimage
+///    HTLCIntercepted -- await preimage offer failed --> CancelHTLC
+///    BuyPreimage -- await preimage decryption --> Preimage
+///    BuyPreimage -- await preimage decryption failure --> CancelHTLC
+///    Preimage -- settle HTLC --> Settled
+///    Preimage -- settle HTLC failure --> Preimage
```

Comment:
I wonder if we should give up eventually? At least there should be a delay so we don't starve other state machines.

Reply:
Added a timeout transition that moves it to the Refund state.